### PR TITLE
Minor documentation fixes related to bulk insert

### PR DESCRIPTION
**Pull request by vishaltelangre**

### CODE REVIEW

File: activerecord/lib/active_record/persistence.rb [hunk 1]
```diff
@@ -222,10 +222,13 @@ def upsert(attributes, returning: nil, unique_by: nil)
       #   new records may considered duplicates according to different criteria. For MySQL,
       #   an upsert will take place if a new record violates _any_ unique constraint. For
       #   Postgres and SQLite, new rows will replace existing rows when the new row has the
-      #   same primary key as the existing row. By defining <tt>:unique_by</tt>, you can supply
-      #   a different key for matching new records to existing ones than the primary key.
+      #   same primary key as the existing row. In case of SQLite, an upsert operation causes
+      #   an insert to behave as an update or a no-op if the insert would violate
+      #   a uniqueness constraint. By defining <tt>:unique_by</tt>, you can supply
+      #   a different unique constraint for matching new records to existing ones than the
+      #   primary key.
```
Comment (vishaltelangre) [inline activerecord/lib/active_record/persistence.rb]: Reference about SQLite's UPSERT behavior - https://www.sqlite.org/lang_UPSERT.html.

> UPSERT is a special syntax addition to INSERT that causes the INSERT to behave as an UPDATE or a no-op if the INSERT would violate a uniqueness constraint. UPSERT is not standard SQL. UPSERT in SQLite follows the syntax established by PostgreSQL. [...]

File: activerecord/lib/active_record/persistence.rb [hunk 2]
```diff
@@ -222,10 +222,13 @@ def upsert(attributes, returning: nil, unique_by: nil)
...
       #
-      #   (For example, if you have a unique index on the ISBN column and use that as
+      #   (For example, if you have a unique index on the isbn column and use that as
```
Comment (vipulnsward) [inline activerecord/lib/active_record/persistence.rb]: There are more occurrences of isbn that differ, can you reconcile them?

**Comment by vipulnsward** [comment]
@vishaltelangre for changes that you do, please add a [ci skip] in the commit message since this is just a doc change and we want to avoid CI runs

File: activerecord/lib/active_record/persistence.rb [hunk 3]
```diff
@@ -173,8 +173,8 @@ def insert!(attributes, returning: nil)
       #     { title: 'Eloquent Ruby', author: 'Russ' }
       #   ])
       #
-      #   # raises ActiveRecord::RecordNotUnique because 'Eloquent Ruby'
-      #   # does not have a unique ID
+      #   # Raises ActiveRecord::RecordNotUnique because 'Eloquent Ruby'
+      #   # does not have a unique id
```
Comment (vipulnsward) [inline activerecord/lib/active_record/persistence.rb]: We can retain the `unique id` change. We use `unique ID` at other places.

File: activerecord/lib/active_record/persistence.rb (hunk 2 above)
Comment (vipulnsward) [reply activerecord/lib/active_record/persistence.rb]: To be precise, lets retain this, its being used same as `ID` vs `id`. Lets keep this usage

File: activerecord/lib/active_record/persistence.rb (hunk 1 above)
Comment (vipulnsward) [inline activerecord/lib/active_record/persistence.rb]: Can you change a line above to: `new records may be considered`. There is a missing `may`

File: activerecord/lib/active_record/persistence.rb (hunk 2 above)
Comment (vishaltelangre) [reply activerecord/lib/active_record/persistence.rb]: I just wanted to change `ISBN column` to `isbn column` when it is explicitly referred as a column.

I think changing `ISBNs` to `isbns` will not look good therefore I will revert this change as well.

File: activerecord/lib/active_record/persistence.rb (hunk 1 above)
Comment (vishaltelangre) [reply activerecord/lib/active_record/persistence.rb]: Done.

File: activerecord/lib/active_record/persistence.rb (hunk 2 above)
Comment (vishaltelangre) [reply activerecord/lib/active_record/persistence.rb]: Done.

File: activerecord/lib/active_record/persistence.rb (hunk 3 above)
Comment (vishaltelangre) [reply activerecord/lib/active_record/persistence.rb]: Makes sense. Retained back to the original.

**Comment by vipulnsward** [comment]
Thanks @vishaltelangre !

**Pull request merged**